src/Maestro/DependencyUpdateErrorProcessor/DependencyUpdateErrorProcessor.cs

```diff
@@ -151,54 +161,85 @@ public async Task ProcessDependencyUpdateErrorsAsync()
         /// <param name="description">Description for the issue body / comment body</param>
         /// <returns></returns>
         private async Task CreateOrUpdateGithubIssueAsync(
-            RepositoryBranchUpdateHistoryEntry updateHistoryError, 
+            UpdateHistoryEntry updateHistoryError, 
             string issueRepo, 
             Func<string, string, bool> shouldReplaceDescription, 
             string description)
         {
-            _logger.LogInformation($"Error Message : '{updateHistoryError.ErrorMessage}' in repository :  '{updateHistoryError.Repository}'");
-            IReliableDictionary<(string repository, string branch),int> gitHubIssueEvaluator =
-                await _stateManager.GetOrAddAsync<IReliableDictionary<(string repository, string branch), int>>("gitHubIssueEvaluator");
+            
             var parseRepoUri = ParseRepoUri(issueRepo);
             IGitHubClient client = await _authenticateGitHubApplicationClient.CreateGitHubClientAsync(parseRepoUri.owner, parseRepoUri.repo);
-            Octokit.Repository repo = await client.Repository.Get(
-                parseRepoUri.owner, 
+            Repository repo = await client.Repository.Get(
+                parseRepoUri.owner,
                 parseRepoUri.repo);
             var issueNumber = new ConditionalValue<int>();
-            using (ITransaction tx = _stateManager.CreateTransaction())
+
+            if (updateHistoryError is RepositoryBranchUpdateHistoryEntry)
```

Comment:
How do you feel about using a pattern matching switch here? https://docs.microsoft.com/en-us/dotnet/csharp/pattern-matching#using-pattern-matching-switch-statements

something like:

```C#
switch (updateHistoryError)
{
  case RepositoryBranchUpdateHistoryEntry repoBranchUpdateHistoryError:
    blah
    break;
  case SubscriptionUpdateHistoryEntry subscriptionUpdateHistoryError  :
    blah;
    break
  default: 
   well.. this is awkward
   break;
}
```

Reply:
that was just not in my arsenal, sure.